src/jsonspec/pointer/stages.py

```diff
@@ -54,10 +54,10 @@ def stage(obj, parent=None, member=None):
     """
     obj = Staged(obj, parent, member)
 
-    if isinstance(obj, dict):
+    if isinstance(obj, MutableMapping):
         for key, value in obj.items():
             stage(value, obj, key)
-    elif isinstance(obj, (list, tuple)):
+    elif isinstance(obj, MutableSequence):
```

Comment:
I'll add some tests to see what works, problem is that strings are of type `Sequence` ...